diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/NullableTypeRule.kt

```diff
@@ -0,0 +1,93 @@
+import com.pinterest.ktlint.core.ast.ElementType.INTEGER_CONSTANT
+import com.pinterest.ktlint.core.ast.ElementType.INTEGER_LITERAL
+import com.pinterest.ktlint.core.ast.ElementType.NULL
+import com.pinterest.ktlint.core.ast.ElementType.NULLABLE_TYPE
+import com.pinterest.ktlint.core.ast.ElementType.OPEN_QUOTE
+import com.pinterest.ktlint.core.ast.ElementType.PROPERTY
+import com.pinterest.ktlint.core.ast.ElementType.REFERENCE_EXPRESSION
+import com.pinterest.ktlint.core.ast.ElementType.STRING_TEMPLATE
+import com.pinterest.ktlint.core.ast.ElementType.TYPE_REFERENCE
+import com.pinterest.ktlint.core.ast.ElementType.USER_TYPE
+import com.pinterest.ktlint.core.ast.ElementType.VAL_KEYWORD
+import org.cqfn.diktat.common.config.rules.RulesConfig
+import org.cqfn.diktat.ruleset.constants.Warnings.NULLABLE_PROPERTY_TYPE
+import org.cqfn.diktat.ruleset.utils.KotlinParser
+import org.cqfn.diktat.ruleset.utils.hasChildOfType
+import org.jetbrains.kotlin.com.intellij.lang.ASTNode
+import org.jetbrains.kotlin.com.intellij.psi.impl.source.tree.CompositeElement
+import org.jetbrains.kotlin.com.intellij.psi.impl.source.tree.LeafPsiElement
+import org.jetbrains.kotlin.com.intellij.psi.tree.IElementType
+
+class NullableTypeRule(private val configRules: List<RulesConfig>) : Rule("nullable-type") {
+
+    private lateinit var emitWarn: ((offset: Int, errorMessage: String, canBeAutoCorrected: Boolean) -> Unit)
+    private var isFixMode: Boolean = false
+
+    override fun visit(node: ASTNode,
+                       autoCorrect: Boolean,
+                       emit: (offset: Int, errorMessage: String, canBeAutoCorrected: Boolean) -> Unit) {
+        isFixMode = autoCorrect
+        emitWarn = emit
+
+        if (node.elementType == PROPERTY)
+            checkProperty(node)
+    }
+
+    private fun checkProperty(node: ASTNode) {
+        if (node.hasChildOfType(VAL_KEYWORD) && node.hasChildOfType(EQ) && node.hasChildOfType(TYPE_REFERENCE) && node.hasChildOfType(NULL)) {
+            NULLABLE_PROPERTY_TYPE.warnAndFix(configRules, emitWarn, isFixMode, "initialize explicitly", node.findChildByType(NULL)!!.startOffset) {
+                findSubstitution(node)
+            }
+        }
+    }
+
+    private fun findSubstitution(node: ASTNode) {
+        val reference = node.findChildByType(TYPE_REFERENCE)!!.findChildByType(NULLABLE_TYPE)!!.findChildByType(USER_TYPE)?.findChildByType(REFERENCE_EXPRESSION)
+                ?: return
+
+        when (reference.text) {
+            "Int", "Short", "Byte" -> replaceValue(node, INTEGER_CONSTANT, INTEGER_LITERAL, "0")
+            "Double", "Float" -> replaceValue(node, FLOAT_CONSTANT, FLOAT_LITERAL, "0.0")
+            "Long" -> replaceValue(node, INTEGER_CONSTANT, INTEGER_LITERAL, "0L")
+            "Char" -> replaceValue(node, CHARACTER_CONSTANT, CHARACTER_LITERAL, "\'\'")
+            "String" -> replaceValueForString(node)
+            "List" -> replaceValueByText(node, "emptyList()")
```

Comment:
these are all types you know in Kotlin? :)

Reply:
No) I haven't finished yet

Reply:
You can open pull request as draft: all CI checks will still be performed, but it will be explicitly marked as not ready for review